pallets/teebag/src/lib.rs

```diff
@@ -640,70 +613,64 @@ impl<T: Config> Pallet<T> {
 		Ok(().into())
 	}
 
-	fn increment_count(worker_type: WorkerType) -> Result<(), DispatchErrorWithPostInfo> {
-		let count = Self::enclave_count(worker_type);
-		ensure!(count < Self::max_enclave_count(worker_type), Error::<T>::MaxEnclaveCountOverflow);
-
-		EnclaveCount::<T>::insert(
-			worker_type,
-			count.checked_add(1u64).ok_or(Error::<T>::MaxEnclaveCountOverflow)?,
-		);
-
-		Ok(())
+	fn add_enclave_identifier(worker_type: WorkerType, who: &T::AccountId) -> Result<(), Error<T>> {
+		EnclaveIdentifier::<T>::try_mutate(worker_type, |v| {
+			ensure!(!v.contains(who), Error::<T>::EnclaveIdentifierAlreadyExist);
+			v.try_push(who.clone()).map_err(|_| Error::<T>::MaxEnclaveIdentifierOverflow)
+		})
 	}
 
-	fn decrement_count(worker_type: WorkerType) -> Result<(), DispatchErrorWithPostInfo> {
-		let count = Self::enclave_count(worker_type);
-		EnclaveCount::<T>::insert(
-			worker_type,
-			count.checked_sub(1u64).ok_or(Error::<T>::MaxEnclaveCountUnderflow)?,
-		);
-
-		Ok(())
+	fn remove_enclave_identifier(
+		worker_type: WorkerType,
+		who: &T::AccountId,
+	) -> Result<(), Error<T>> {
+		EnclaveIdentifier::<T>::try_mutate(worker_type, |v| {
+			ensure!(v.contains(who), Error::<T>::EnclaveIdentifierNotExist);
+			v.retain(|e| e != who);
+			Ok(())
+		})
 	}
 
-	pub fn add_enclave(sender: &T::AccountId, enclave: &Enclave) -> DispatchResultWithPostInfo {
+	pub fn add_enclave(sender: &T::AccountId, enclave: &Enclave) -> Result<(), Error<T>> {
 		match EnclaveRegistry::<T>::get(sender) {
 			Some(old_enclave) => {
 				if old_enclave.worker_type != enclave.worker_type {
-					// a tricky situation - we are re-registering the enclave with a different
-					// worker type
-					Self::decrement_count(old_enclave.worker_type)?;
-					Self::increment_count(enclave.worker_type)?;
+					// a tricky situation - an old enclave is registered with a different
```

Comment:
I think the core question is shall we allow registering a different worker type using the same parachain account - as if we are re-using the same machine with previously sealed parachain account to re-reigster itself (I can only imagine this in test).

Either way is fine for me - can you think of any cases where we heavily prefer A over B?

Reply:
I think I would prefer to have a clear rule 1 parachain account - 1 worker type.
It's relatively cheap to create new account compared to how it can potentially make things complex (now or future).

Reply:
Sure I'll change it